lib/node/nodes/buffer.js

```diff
@@ -122,17 +138,11 @@ Buffer.prototype.isolinesDissolvedQuery = function() {
 };
 
 Buffer.prototype.checkLimits = function(context, callback) {
-    var defaultLimit;
-    var limitName;
-    var limit;
+    var limits = this.dissolved ? LIMITS_DISSOLVED : LIMITS;
 
-    if (this.dissolved) {
-        limitName = 'maximumNumberOfRowsDissolved';
-        defaultLimit = 100000;
-    } else {
-        limitName = 'maximumNumberOfRows';
-        defaultLimit = 1000000;
-    }
-    limit = context.getLimit(TYPE, limitName, defaultLimit, 'too many source rows');
-    limits.limitInputRows(this, 'source', context, limit, callback);
+    Checks.check(this, context, [{
+        checker: Checks.limitInputRows,
```

Comment:
What I like about `this.limits.checkingOp(...)` vs `Checks.check(..., [{...definitions...}])` is how it simplifies some things (we don't need a check for input rows, and passing the `input` parameter since we have a check for output rows).